p2p/src/p2p_state.rs

```diff
@@ -491,38 +491,56 @@ type OptionalCallback<T> = Option<Callback<T>>;
 
 #[derive(Serialize, Deserialize, Debug, Clone, Default)]
 pub struct P2pCallbacks {
+    /// Callback for [`P2pChannelsTransactionAction::Libp2pReceived`]
     pub on_p2p_channels_transaction_libp2p_received:
         OptionalCallback<Box<MinaBaseUserCommandStableV2>>,
+    /// Callback for [`P2pChannelsSnarkJobCommitmentAction::Received`]
     pub on_p2p_channels_snark_job_commitment_received:
         OptionalCallback<(PeerId, Box<SnarkJobCommitment>)>,
 
+    /// Callback for [`P2pChannelsSnarkAction::Received`]
     pub on_p2p_channels_snark_received: OptionalCallback<(PeerId, Box<SnarkInfo>)>,
+    /// Callback for [`P2pChannelsSnarkAction::Libp2pReceived`]
     pub on_p2p_channels_snark_libp2p_received: OptionalCallback<(PeerId, Box<Snark>)>,
 
+    /// Callback for [`P2pChannelsBestTipAction::RequestReceived`]
     pub on_p2p_channels_best_tip_request_received: OptionalCallback<PeerId>,
 
+    /// Callback for [`P2pDisconnectionAction::Finish`]
     pub on_p2p_disconnection_finish: OptionalCallback<PeerId>,
 
-    pub on_p2p_connection_outgoing_error:
-        OptionalCallback<(RequestId<RpcIdType>, P2pConnectionOutgoingError)>,
-    pub on_p2p_connection_outgoing_success: OptionalCallback<RequestId<RpcIdType>>,
+    /// Callback for [`P2pConnectionOutgoingAction::Error`]
+    pub on_p2p_connection_outgoing_error: OptionalCallback<(RpcId, P2pConnectionOutgoingError)>,
+    /// Callback for [`P2pConnectionOutgoingAction::Success`]
```

Comment:
@0xMimir that would be `P2pConnectionOutgoingAction::Init` right? Basically, this callback will be called as a result of the `Init` action.

So on the caller which now has:

```rust
store.dispatch(P2pConnectionOutgoingAction::Init {
    opts,
    rpc_id: Some(rpc_id),
});
```

if we were to add the callbacks, we would have:

```rust
store.dispatch(P2pConnectionOutgoingAction::Init {
    opts,
    rpc_id: Some(rpc_id),
    on_success: ....this callback....,
    on_error: ...the error callback...,
});

Reply:
@0xMimir I see this still contains the previous comments, so let me clarify. The idea is that it should mention the original action to which it relates (from the point of view of the caller). Like in the example above, the caller will never dispatch `Success` or `Error`, he will dispatch `Init`, so this is a callback for `Init`. But it is no big deal, just a nice-to-have to make it easier to modification I mentioned above in the future.